torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1259,6 +1260,41 @@ def generator():
 
     return list(generator())
 
+def sample_inputs_batch_norm(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    make_arg_without_requires_grad = partial(make_tensor, device=device, dtype=dtype, requires_grad=False)
+
+    # Ordered as: input shape, kwargs for training, momentum, eps
+    cases: Tuple[Tuple[int], dict] = (  # type: ignore[assignment]
+        ((S, S, S), {'training': True, 'momentum': 0.5, 'eps': 0.6}),
+        ((3, 2, 4), {'training': False, 'momentum': -1.2}),
+        ((3, 1), {'training': True, 'momentum': 0.0}),
+        ((2, 1), {'training': False}),
+        ((3, 2, 3, 4), {'training': True, 'momentum': -1.0, 'eps': 0.5}),
+        ((3, 2, 3, 4), {'training': False, 'momentum': -1.0, 'eps': 0.5}),
+        ((2, 1), {})
+    )
+
+    def generator():
+        for input_shape, kwargs in cases:
+            # args: running mean, running var, weight and bias should necessarily be of shape: (channels,)
+            channels = input_shape[1]
+            yield SampleInput(
+                make_arg(input_shape),
+                args=(
+                    make_arg_without_requires_grad(channels, low=0),  # running mean
+                    make_arg_without_requires_grad(channels, low=0),  # running var
+                    make_arg(channels),  # weight
+                    make_arg(channels)   # bias
+                ),
+                kwargs=kwargs
+            )
+        # Test case for no optional kwargs
+        # running_mean and running_var are required in evaluation mode (training: False) but not in training mode
+        yield SampleInput(make_arg((1, 2, 3)), args=(None, None), kwargs={'training': True})
```

Comment:
Hmm, It turns out that weight and bias can also be None. Would it be worth testing the following additional cases?
- input, running_mean, running_var, weight, None, training=True
- input, running_mean, running_var, None, bias, training=False
- input, running_mean, running_var, None, None, training=False

Reply:
Agreed, good suggestion. I've added these cases.